code/modules/fish/fishtank.dm

```diff
@@ -335,6 +344,7 @@
 		var/count = length(fish_types[key])
 		fish_types_input += list("[initial(fish_type.fish_name)][count > 1 ? " (x[count])" : ""]" = fish_types[key])
 	var/caught_fish = input("Select a fish to catch.", "Fishing") as null|anything in fish_types_input		//Select a fish from the tank
+	current_fisher = null
```

Comment:
I see a possible issue here, what if a player _doesn't_ close this input menu. What if they walk halfway across the station and go afk or walk away from their keyboard? Current_fisher will never null until that input is given.

Reply:
Just tested this, and yes, you can walk away and leave the fishtank thus locking the fish tank. I'm going to have a fiddle and see what else I can come up with.